Automattic/jetpack
# [Boost] Use snake_case on tracks event props

Related: p1677079092758209-slack-C016BBAFHHS

Our Image Guide is using camelCase event props for tracks events, which are invalid. This PR fixes the props passed from all Image Guide analytics calls to use snake_case.

It also moves the TracksEventProps type out of global, where it doesn't really belong. :) 

## Proposed changes:
* Use snake_case when reporting tracks events.

### Other information:

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
It fixes stuff we were already trying to track.

## Testing instructions:
* Run the image guide
* Make sure that tracks events are reported in camel_case (check the Network tab).
i.e.:
![image](https://user-images.githubusercontent.com/1369626/220850689-b457b424-25ce-4773-9314-03006bff4766.png)
